BasicMusicPlayer/app/src/main/java/com/example/basicmusicplayer/PlayerActivity.kt

```diff
@@ -160,81 +112,142 @@ class PlayerActivity : AppCompatActivity() {
+                    initializeActivity()
+                    return@launch
+                }
 
-    // plays the stream of the radi
-    private fun playRadio() {
-        val wxycURL = "http://audio-mp3.ibiblio.org:8000/wxyc-alt.mp3"
-        loadingIndicator.visibility = View.VISIBLE
-        //media player created and initialized
-        mediaPlayer = MediaPlayer().apply {
-            setAudioAttributes(
-                AudioAttributes.Builder()
-                    .setUsage(AudioAttributes.USAGE_MEDIA)
-                    .setContentType(AudioAttributes.CONTENT_TYPE_MUSIC)
-                    .build()
-            )
-            setDataSource(wxycURL)
-            //set to handle event when audio is prepared
-            setOnPreparedListener { mp ->
-                mp.start()
-                loadingIndicator.visibility = View.GONE // Hide the loading indicator
-                Toast.makeText(applicationContext, "Audio started playing", Toast.LENGTH_LONG)
-                    .show()
-            }
-            //initiates prep process
-            prepareAsync()
-        }
-    }
+                // this section compares the first 7 entries of the current list and an updated list
+                // if there are differences the first 7 entries will be updated
 
+                // fetches last 7 updated playlist values
+                var updatedSubList = playlistManager.fetchLittlePlaylist()
+                if (updatedSubList.isNullOrEmpty()) {
+                    // switch to null or empty
+                    return@launch
+                }
+                if (updatedSubList.size != UPDATE_UPPER_VALUE) {
+                    updatedSubList = updatedSubList.subList(0, UPDATE_UPPER_VALUE)
+                }
 
-    // ensures it releases the mediaPlayer when app is closed?
-    override fun onStop() {
-        super.onStop()
-        releaseMediaPlayer()
-    }
+                // fetches last 7 current playlist values
+                val currentSubList = playlistDetailsList.subList(0, UPDATE_UPPER_VALUE)
 
-    private fun releaseMediaPlayer() {
-        mediaPlayer?.release()
-        mediaPlayer = null
-    }
+                // bool values to keep track of update type
+                var editPlaylist = false
+                var newEntry = false
+
+                //  checks if the lists are the same
+                if (!compareLists(updatedSubList, currentSubList)) {
+                    // this determined if the two sublists are different.  now we need to check if
+                    // an entry was added or there was an edit to the playlist (with no added entry)
 
+                    // checks if the content in the lists are the same i.e. just an edit in order
+                    if (compareListContent(updatedSubList, currentSubList)) {
+                        editPlaylist = true
+                    }
+                    // new entry
+                    else {
+                        newEntry = true
```

Comment:
If `editPlaylist` and `newEntry` are mutually exclusive, would it make sense to just use one variable?